### ISSUE TITLE
az command not executable

### ISSUE THREAD

**Issue by abassett-nh** [Azure/azure-cli#17460]
## Describe the bug

**Command Name**
`az login`

**Errors:**
`-bash: /home/ubuntu/aztest/bin/az: Permission denied`

## To Reproduce:
```
python3 -m virtualenv aztest
source aztest/bin/activate
pip install azure-cli
az login
```
## Expected Behavior
az should be installed with the executable bit set:

```
ll /home/ubuntu/csitest/bin/az
-rw-r----- 1 ubuntu ubuntu 482 Mar 25 11:15 /home/ubuntu/functionvenv/bin/az
```

setting the bit fixes it
```
chmod +x  /home/ubuntu/functionvenv/bin/az
az login
```
## Environment Summary
```
Linux-5.4.0-1043-azure-x86_64-with-glibc2.29
Python 3.8.5
Installer: PIP

azure-cli 2.21.0

```

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by usrme** [comment]
I'm not affiliated with Microsoft nor the Azure CLI, but I just tested the exact same steps as you provided on my own machine and was unable to reproduce this :thinking:

What is the output of the `umask` command?

**Comment by jiasli** [comment]
Thanks @usrme. I also tested with a clean Ubuntu 20.04 VM and couldn't repro it.

**Comment by usrme** [comment]
Yeah, it seems like a super specific client-side issue. I would go ahead and close this if I were you.

**Comment by abassett-nh** [comment]
```
umask
0027
```

fwiw i'm using a cis hardened ubuntu image from the azure marketplace: "CIS Ubuntu Linux 20.04 LTS Benchmark L1"

This is the only python package I've encountered with this issue. The others I've installed on the same node all correctly mark their executables executable.
